home/views.py

```diff
@@ -39,9 +39,17 @@ def new_question(request):
     if request.method == 'POST':
         questForm = QuestionForm(request.POST)
         if questForm.is_valid():
-            questForm = questForm.save(commit=False)
-            questForm.profile = request.profile
-            questForm.save()
+            tags_array = [t.strip().lower() for t in questForm.cleaned_data['tags_'].split(',')]
+            if(not questForm.data['tags_']):
+                questForm = questForm.save(commit=False)
+                questForm.profile = request.profile
+                questForm.save()
+            elif(Tag.check_tag_array(tags_array)):
+                questForm = questForm.save(commit=False)
+                questForm.profile = request.profile
+                questForm.save()
```

Comment:
@avivz450 
You have duplicate code here . 
the following code :
```
questForm = questForm.save(commit=False)   
questForm.profile = request.profile
questForm.save()
```
appears in both if and elif sections. 
I think you can execute this code and only after it write the if section

Reply:
Not exactly... we need to save the question only if:

- the tags field is empty - then we save only the question.
- the tags field isn't empty and valid - then we save the question and then connect the tags to the question object.

When we wrote it, we noticed the duplication but at the time couldn't exactly think of a better way. 
Do you see a way to remove the duplication and keep the logic?

Reply:
do this 3 lines 
`questForm = questForm.save(commit=False)   
questForm.profile = request.profile
questForm.save()`
without condition (or with condition of both) and then do the 2 other lines only if there is tags.

Reply:
We noticed the duplication of this code, but I have to keep it like that because I am going to add to the PR of the "Alert messages" an "else" option and it will display an error message.
Therefore, even if I will do what you said, in the next PR I will have to return the code to that way.

Reply:
I think we have to have a condition, otherwise we might save a question with unvalid tags.
`
tags_status = Tag.check_tag_array(tags_array)
if(not questForm.data['tags_'] or tags_status):
    questForm = questForm.save(commit=False)
    questForm.profile = request.profile
    questForm.save()            
    if(tags_status):
        questionInstane = Question.objects.get(id=questForm.id)
        questionInstane.add_tags_to_question(tags_array)
`
This is a way to avoid the duplication of the 2 lines but then there is a duplication in the if.
What do you think? @danitLevi @idokk1

Reply:
@rebeccaTubman
looks good

Reply:
made that change